src/configuration.js

```diff
@@ -1,12 +1,29 @@
+const cosmiconfig = require('cosmiconfig');
+import { readSync, readFileSync } from 'fs';
+import getGraphQLProjectConfig from 'graphql-config';
+import path from 'path';
+
 import defaultRules from './rules/index.js';
 import JSONFormatter from './formatters/json_formatter.js';
 import TextFormatter from './formatters/text_formatter.js';
-import getGraphQLProjectConfig from 'graphql-config';
-import { readSync, readFileSync } from 'fs';
 
 export class Configuration {
-  constructor(options, stdinFd) {
-    this.options = options;
+  /*
+    options:
+      - configDirectory: path to begin searching for config files
+      - except: [string array] blacklist rules
+      - format: (required) `text` | `json`
+      - only: [string array] whitelist rules
+      - schemaFileName: [string] file to read schema from
+      - stdin: [boolean] pass schema via stdin?
+  */
+  constructor(options = {}, stdinFd = null) {
+    const defaultOptions = { format: 'text' };
+    const configOptions = loadOptionsFromConfig(
+      options.configDirectory || process.cwd()
```

Comment:
So it turns out you can either pass a config directory (full path) or a search path (relative to `process.cwd()`), so I don't even need to pass process.cwd() all the way down.  See: https://github.com/davidtheclark/cosmiconfig/blob/master/lib/createExplorer.js#L52